src/main/java/mainpack/AbstractSubject.java

```diff
@@ -0,0 +1,31 @@
+package mainpack;
```

Comment:
Also, i think it's better to divide classes into different packages:
mainpack - only for MainRun
mainpack.subject - for subjects
and so on, feel free to create other packages

Reply:
It helps to orginize your workspace and also to separate your code logically